### ISSUE TITLE
Incompilable code when inheriting classes with private or protected modifier

### ISSUE THREAD

**Issue by genuinelucifer** [mono/CppSharp#493]
"error C2243: 'type cast' : conversion from 'DerivedClass *' to 'BaseClass *' exists, but is inaccessible" is generated in C# code if a class is inherited except publicly.
Isn't it supported already??
To recreate just inherit from another class using 'protected' or 'private' modifier.

**Comment by ddobrev** [comment]
We have indeed not bothered to support this yet. What do you need it for?

**Comment by tritao** [comment]
We should still fix it though if it generates broken code.

**Comment by ddobrev** [comment]
I agree but it's neither as important as @genuinelucifer 's other tasks nor as the task he requested to work on (https://github.com/mono/CppSharp/issues/96).

**Comment by genuinelucifer** [comment]
Well, I was just trying to write a test case and by mistake forgot to put
the 'public' keyword and the inheritance defaulted to private.
Then I saw the broken.code generated so I thought to open.the issue.
Obviously it is not important at this point of time. But, I opened it so
that it may be fixed sometime in near future.

**Comment by ddobrev** [comment]
Of course, you did right by sending this report. Thank you. We'll definitely fix it at some point.
